lessons/cryptography/Encoding/encoding.md

```diff
@@ -0,0 +1,118 @@
+# [Cryptography] Encoding
+*Learn about encoding, a way to reformat information.*
+
+![banner alt text](.rsrc/banner.png)
+
+# Objectives
+- ASCII
+- Unicode
+- Base64
+
+# Introduction
+Encoding refers to converting information into a specific form. Although encoding and encryption may sound similar, these two terms should not be confused. Encryption refers to converting information into a different form in order to keep information secret, such that decryption would be difficult without special knowledge. Encoding also converts information, but makes no attempt to prevent other people from obtaining the original message. In fact, encoding and decoding are simply opposite processes. Encoding is primarily used to ensure efficient transmission or storage of information. 
+
+The most common types of encoding include ASCII, unicode, and base64. A tool called [CyberChef](https://gchq.github.io/CyberChef/) can be useful for encoding and decoding messages. 
+
+1. ASCII
+2. Unicode 
+3. Base64
+
+# Lesson
+## ASCII
+ASCII stands for American Standard Code for Information Interchange, and is a character encoding standard. Characters that can be encoded include uppercase letters, lowercase letters, digits, and many punctuation marks. The standard ASCII table includes 128 encoded characters using a 7-bit binary code. There is also an extended ASCII table (the modern standard) that includes 256 characters, which is the max number of characters that can be encoded with an 8-bit binary code. The table below shows both the basic and extended ASCII tables. 
```

Comment:
I think this is going to feel like a lot for the students. Maybe move the example on line 28 up to here so that we don't lose them. Open to ideas from you though.

Reply:
If you think they don't need to know this, I'm cool with just removing most of it. One of the reasons I wanted to point it out is b/c I saw references to 7-bit vs 8-bit ASCII in some places and they might run into issues later, but I agree it's a lot. Do you think it's better to put this extra info at the end of the ASCII section instead? or just take it out?

Reply:
Its good to mention, 8-bit is a thing albeit uncommon. Yeah probably move it to the end of the ASCII section though. And bringing up that line 28 material should help it become more cohesive.